### PR TITLE
Add multi-select mode to word list with bulk delete

### PR DESCRIPTION
**Pull request by Vladyslav-Soldatenko**
Implement selection mode for the word list screen, allowing users to select multiple vocabulary items and perform bulk operations.

## Summary
This change adds a multi-select UI pattern to the word list, enabling users to long-press a word to enter selection mode, toggle individual selections, and bulk-delete selected items. Selection state is managed in `WordListViewModel` and persists across configuration changes but resets when navigating away from the screen.

## Key Changes

- **Selection state management**: Added `SelectionState` data class to `WordListViewModel` with methods to enter/exit selection mode, toggle individual selections, select/deselect all, and delete selected items
- **UI updates to `WordListScreen`**:
  - Long-press on a word row enters selection mode with that word pre-selected
  - Tap toggles selection when in selection mode; long-press is disabled to avoid re-entering
  - Checkboxes appear on rows only in selection mode
  - Header shows selection count instead of title when active
  - Back button exits selection mode instead of navigating back
  - Overflow menu in header shows "Select all" / "Deselect all" and "Delete" options
  - Per-row overflow menus are hidden in selection mode
  - Selected rows are highlighted with a subtle background tint
  - System back press exits selection mode
- **Bulk delete confirmation**: New `BulkDeleteWordsDialog` shows count and requires confirmation before deletion
- **Repository layer**: Added `deleteVocabularyItems()` to support bulk deletion; refactored singular `deleteVocabularyItem()` to delegate to it
- **Database layer**: Extended `VocabularyDao` and `UndoSnapshotDao` with batch delete methods
- **Comprehensive test coverage**: 
  - 30+ new UI tests covering selection interactions, header state, menu behavior, and edge cases
  - 10+ new ViewModel tests for selection state transitions
  - Integration test verifying selection state resets on navigation
  - Repository tests for bulk delete with undo snapshot cleanup
- **Localization**: Added "Select all" and "Deselect all" strings to all supported languages (English, Russian, Ukrainian, Spanish, French, Italian, Chinese)

## Implementation Details

- Selection mode is scoped to the `WordListViewModel` instance, which is tied to the word list's `NavBackStackEntry`. This ensures selection state resets when navigating away and back, even with bottom nav's `popUpTo(saveState = true)` behavior.
- The "Select all" menu item respects the current search filter—it only selects visible words, not hidden ones.
- The menu intelligently flips to "Deselect all" when all displayed words are selected, or when the selection is a superset of displayed words (e.g., user selected all, then filtered).
- Bulk delete only removes items still present in the word list, safely handling race conditions where selected IDs may no longer exist.

### CODE REVIEW

File: app/src/androidTest/java/com/procrastilearn/app/navigation/WordListSelectionNavigationResetTest.kt [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+ * WordListScreen's selection mode lives entirely in WordListViewModel, which is scoped to
+ * WordList's own NavBackStackEntry. Since WordList is only ever entered via a plain
+ * navController.navigate(Screen.WordList.route) (no restoreState), each visit should get a
+ * fresh backstack entry - and therefore fresh state - even after the entry is popped off
+ * (and possibly saved) by the bottom nav's popUpTo(saveState = true) + restoreState = true
+ * dance on other tabs. This test stands in a locally-`remember`ed boolean for the real
+ * ViewModel's selectionState to verify that assumption without needing Hilt test infra
+ * (this codebase has none), mirroring the bare-NavHost pattern already used by
+ * BottomNavigationBarTest.
+ */
```
Comment (Vladyslav-Soldatenko) [inline app/src/androidTest/java/com/procrastilearn/app/navigation/WordListSelectionNavigationResetTest.kt]: remove comment

File: app/src/androidTest/java/com/procrastilearn/app/navigation/WordListSelectionNavigationResetTest.kt [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+    private companion object {
+        const val APPS_TEXT = "apps_screen"
+        const val ADD_WORD_TEXT = "add_word_screen"
+        const val DOJO_TEXT = "dojo_screen"
+        const val SETTINGS_TEXT = "settings_screen"
+        const val SELECTION_ACTIVE_TEXT = "selection_active"
+        const val SELECTION_INACTIVE_TEXT = "selection_inactive"
+        const val ENTER_SELECTION_BUTTON = "enter_selection"
+    }
```
Comment (Vladyslav-Soldatenko) [inline app/src/androidTest/java/com/procrastilearn/app/navigation/WordListSelectionNavigationResetTest.kt:86]: can we reuse existing routes obj, not to duplicate?
